### PR TITLE
Add `MaterialTag`

### PR DESCRIPTION
**Pull request by tal5**

### CODE REVIEW

File: src/main/java/com/denizenscript/clientizen/objects/MaterialTag.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import net.minecraft.block.BlockState;
+import net.minecraft.item.Item;
+import net.minecraft.registry.Registries;
+import net.minecraft.state.State;
+import net.minecraft.util.Identifier;
+
+public class MaterialTag implements ObjectTag, Adjustable {
+
+    public BlockState state;
+    public Item item;
```
Comment (acikek) [inline src/main/java/com/denizenscript/clientizen/objects/MaterialTag.java:22]: There should be a comment here explaining why both of these fields are present.

File: src/main/java/com/denizenscript/clientizen/util/Utilities.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.denizenscript.clientizen.util;
+
+import net.minecraft.util.Identifier;
+
+public class Utilities {
+
+    public static String stringifyIdentifier(Identifier identifier) {
```
Comment (acikek) [inline src/main/java/com/denizenscript/clientizen/util/Utilities.java:7]: `idToString`, maybe?

File: src/main/java/com/denizenscript/clientizen/tags/ClientTagBase.java [hunk 3]
```diff
@@ -60,5 +63,18 @@ public void register() {
...
             return DenizenCore.serverFlagMap.doFlagMapTag(attribute);
         });
+        tagProcessor.registerTag(MaterialTag.class, "facing_material", (attribute, object) -> {
+            MinecraftClient client = MinecraftClient.getInstance();
+            if (client.crosshairTarget instanceof BlockHitResult blockHitResult) {
+                return new MaterialTag(client.world.getBlockState(blockHitResult.getBlockPos()));
+            }
+            return null;
+        });
+        tagProcessor.registerMechanism("modifyblock", false, MaterialTag.class, (object, mechanism, input) -> {
```
Comment (acikek) [inline src/main/java/com/denizenscript/clientizen/tags/ClientTagBase.java:73]: maybe `modify_block`? I don't know if this is temporary, but it the conventions seem to be to use underscores for spaces in tags but keep commands without spaces

File: src/main/java/com/denizenscript/clientizen/objects/properties/PropertyRegistry.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    private static <T extends Property<?>> T[] getHandledProperties(Class<? extends MaterialMinecraftProperty<T, ?>> propertyClass) {
+        try {
+            Field handledField = propertyClass.getDeclaredField("handledProperties");
+            return (T[]) handledField.get(null);
+        }
+        catch (IllegalAccessException | NoSuchFieldException e) {
+            Debug.echoError("Invalid handled properties array for property class '" + DebugInternals.getClassNameOpti(propertyClass) + "': " + e.getMessage() + "!");
+        }
+        return null;
+    }
```
Comment (acikek) [inline src/main/java/com/denizenscript/clientizen/objects/properties/PropertyRegistry.java:64]: I think you should remove all of these methods. Reflection is not needed here. Is there a way you can just pass in the required elements from the custom property class into `registerPropertyGetter`? For example, `MaterialLevel`'s register call would be `registerPropertyGetter("level", MaterialIntProperty::new, MaterialLevel.handledProperties, MaterialLevel.class);` Trying to simplify this with these methods yields diminishing returns, imo

File: src/main/java/com/denizenscript/clientizen/objects/MaterialTag.java (hunk 1 above)
Comment (tal5) [reply src/main/java/com/denizenscript/clientizen/objects/MaterialTag.java:22]: Added

File: src/main/java/com/denizenscript/clientizen/util/Utilities.java (hunk 2 above)
Comment (tal5) [reply src/main/java/com/denizenscript/clientizen/util/Utilities.java]: That's a better name lol, ty

File: src/main/java/com/denizenscript/clientizen/tags/ClientTagBase.java (hunk 3 above)
Comment (tal5) [reply src/main/java/com/denizenscript/clientizen/tags/ClientTagBase.java:73]: It's just for testing, we should have a proper modifyblock command in the future

File: src/main/java/com/denizenscript/clientizen/objects/properties/PropertyRegistry.java (hunk 4 above)
Comment (tal5) [reply src/main/java/com/denizenscript/clientizen/objects/properties/PropertyRegistry.java]: Fair point, it is a little more verbose but I suppose it does clean it up

**Pull request merged**